### PR TITLE
DOC: Spell check the code in Examples

### PR DESCRIPTION
**Pull request by dave3d**
## PR Checklist
- [x] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [x] Added test (or behavior not changed)
- [x] Updated API documentation (or API not changed)
- [x] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [x] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [x] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

File: Examples/IO/DicomSeriesReadImageWrite.cxx [hunk 1]
```diff
@@ -44,7 +44,7 @@
 //  Probably the most common representation of datasets in clinical
 //  applications is the one that uses sets of DICOM slices in order to compose
 //  tridimensional images. This is the case for CT, MRI and PET scanners. It
-//  is very common therefore for image analysists to have to process
+//  is very common therefore for image analysts to have to process
```
Comment (dzenanz) [inline Examples/IO/DicomSeriesReadImageWrite.cxx:47]: `analysis` seems just as valid replacement.

File: Examples/RegistrationITKv4/ImageRegistration7.cxx [hunk 2]
```diff
@@ -50,7 +50,7 @@
 // scaling.
 //
 //
-// \index{itk::Simularity2DTransform}
+// \index{itk::Similarity2DTransform}
```
Comment (dzenanz) [inline Examples/RegistrationITKv4/ImageRegistration7.cxx:53]: Hopefully, these will fix the hyperlinks in the docs.

File: Examples/Statistics/ImageHistogram4.cxx [hunk 3]
```diff
@@ -156,7 +156,7 @@ main(int argc, char * argv[])
   // using the \code{SetHistogramBinMinimum()} and
   // \code{SetHistogramBinMaximum()} methods.
   //
-  // Software Guide : EndLatexex
```
Comment (dzenanz) [inline Examples/Statistics/ImageHistogram4.cxx:159]: This probably caused some problems during software guide generation. Good to have it fixed!

**Review by dzenanz — APPROVED** [review]

**Review by hjmjohnson — APPROVED** [review]

File: Examples/Statistics/ImageHistogram4.cxx (hunk 3 above)
Comment (jhlegarreta) [reply Examples/Statistics/ImageHistogram4.cxx:159]: These doc fixes are awesome :100:. Thanks Dave.

I had a look to the SW Guide out of curiosity, and the block was being processed correctly; even the figure name fix in https://github.com/InsightSoftwareConsortium/ITK/pull/4008/files#diff-f001c5fd30fbde05e83c609f5cc7c143b5ef4f87c85ea91e25cabd0c32ef8df7R275 or a close typo in https://github.com/InsightSoftwareConsortium/ITK/pull/4008/files#diff-f001c5fd30fbde05e83c609f5cc7c143b5ef4f87c85ea91e25cabd0c32ef8df7R257 are not present in the [ITK SW Guide version in itk.org](https://itk.org/ItkSoftwareGuide.pdf) (April 16, 2021). Weird.

**Pull request merged**